### PR TITLE
Handle external files with File instead

### PR DESCRIPTION
**Pull request by plankp**
Fixes #3484 

Probably obvious, but the replay will not play if you do not have the beatmap imported already.
Also, you are dragging individual `.osr` files in, not the entire zipped archive.

### CODE REVIEW

File: osu.Game/Rulesets/Scoring/ScoreStore.cs [hunk 1]
```diff
@@ -49,8 +50,24 @@ public void Import(params string[] paths)
...
         {
-            using (Stream s = storage.GetStream(Path.Combine(replay_folder, replayFilename)))
-                return new DatabasedLegacyScoreParser(rulesets, beatmaps).Parse(s);
+            Stream stream;
+            if (File.Exists(replayFilename))
+            {
+                // Handle replay with File since it is outside of storage
+                stream = File.OpenRead(replayFilename);
+            }
+            else if (storage.Exists(Path.Combine(replay_folder, replayFilename)))
```
Comment (peppy) [inline osu.Game/Rulesets/Scoring/ScoreStore.cs]: This fallback logic is really weird. Just remove it IMO.

**Review by peppy — CHANGES_REQUESTED** [review]

File: osu.Game/Rulesets/Scoring/ScoreStore.cs (hunk 1 above)
Comment (plankp) [reply osu.Game/Rulesets/Scoring/ScoreStore.cs]: Remove the `storage.Exists` part?

Comment (peppy) [reply osu.Game/Rulesets/Scoring/ScoreStore.cs]: yes. it is never used.

**Comment by plankp** [comment]
Will the field `storage` be used in the future? inspectcode is complaining about the fact that it isn't used.

**Comment by peppy** [comment]
It can be removed for the time being.

File: osu.Game/Rulesets/Scoring/ScoreStore.cs [hunk 2]
```diff
@@ -49,8 +50,24 @@ public void Import(params string[] paths)
 
         public Score ReadReplayFile(string replayFilename)
         {
-            using (Stream s = storage.GetStream(Path.Combine(replay_folder, replayFilename)))
-                return new DatabasedLegacyScoreParser(rulesets, beatmaps).Parse(s);
+            Stream stream;
+            if (File.Exists(replayFilename))
```
Comment (peppy) [inline osu.Game/Rulesets/Scoring/ScoreStore.cs]: Reverse conditional and return early.

File: osu.Game/Rulesets/Scoring/ScoreStore.cs [hunk 3]
```diff
@@ -49,8 +50,24 @@ public void Import(params string[] paths)
...
+            {
+                // Handle replay with File since it is outside of storage
+                stream = File.OpenRead(replayFilename);
+            }
+            else if (storage.Exists(Path.Combine(replay_folder, replayFilename)))
+            {
+                stream = storage.GetStream(Path.Combine(replay_folder, replayFilename));
```
Comment (peppy) [inline osu.Game/Rulesets/Scoring/ScoreStore.cs]: move inside `using`

**Review by peppy — DISMISSED** [review]

**Comment by plankp** [comment]
Fixed changes

**Review by smoogipoo — APPROVED** [review]
This is fine for now 👍

**Pull request merged**